### PR TITLE
CClient: fixing ESP32 platform issues.

### PR DESCRIPTION
**Pull request by oopsmonk**
* Disable PoW on ESp32.
* Fix overflow issue in `json_array_to_uint64`

### CODE REVIEW

File: cclient/api/core/attach_to_tangle.c [hunk 1]
```diff
@@ -66,6 +67,10 @@ retcode_t iota_client_attach_to_tangle(iota_client_service_t const* const servic
         goto done;
       }
     }
+#else
+    log_info(client_core_logger_id, "[%s:%d] Local PoW is not supported on ESP32.\n", __func__, __LINE__);
```
Comment (jserv) [inline cclient/api/core/attach_to_tangle.c]: PoW might be absent not only for ESP32 but also for low-end devices such as Arm Cortex-M based SoC. Would it be better if there is a configurable switch and/or runtime info pointing PoW availability?

File: common/model/transfer.c [hunk 2]
```diff
@@ -21,6 +21,10 @@ retcode_t transfer_message_set_trytes(transfer_t* tf, tryte_t const* const tryte
 }
 
 retcode_t transfer_message_set_string(transfer_t* tf, char const* const str) {
+  if (strlen(str) == 0) {
```
Comment (jserv) [inline common/model/transfer.c]: You can simply rewrite as `if (!str) {`, which is shorter and faster.

Comment (oopsmonk) [reply common/model/transfer.c]: this check is both for NULL pointer and string length. cannot use `if (!str) {` in this case.

Comment (jserv) [reply common/model/transfer.c]: Use `if (!str || !*str)` instead.

Comment (oopsmonk) [reply common/model/transfer.c]: don't see a reason for it, but here might misleading and don't need to call `strlen` twice.

File: cclient/api/core/attach_to_tangle.c (hunk 1 above)
Comment (oopsmonk) [reply cclient/api/core/attach_to_tangle.c]: it should depend on the platform. the user already has the choice of performing remote/local PoW at runtime.

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**